test/lib/token.test.js

```diff
@@ -0,0 +1,117 @@
+
+describe('Token Model', () => {
+    const password = 'password';
+    let BaseModel;
+    let TokenModel;
+    let ironMock;
+    let datastore;
+    let createConfig;
+    let token;
+
+    before(() => {
+        mockery.enable({
+            useCleanCache: true,
+            warnOnUnregistered: false
+        });
+    });
+
+    beforeEach(() => {
+        datastore = {
+            update: sinon.stub()
+        };
+        datastore.update.resolves({});
+
+        ironMock = {
+            seal: sinon.stub(),
+            unseal: sinon.stub(),
+            defaults: {}
+        };
+        mockery.registerMock('iron', ironMock);
+
+        // eslint-disable-next-line no-global-assign
+        Date = function Date() {
+            this.toISOString = () => 'updated timestamp';
+        };
+
+        // eslint-disable-next-line global-require
+        BaseModel = require('../../lib/base');
+
+        // eslint-disable-next-line global-require
+        TokenModel = require('../../lib/token');
+
+        createConfig = {
+            datastore,
+            userId: 12345,
+            value: 'A_SECRET_TOKEN_VALUE',
+            id: 6789,
+            name: 'Mobile client auth token',
+            description: 'For the mobile app',
+            lastUsed: '2017-05-10T01:49:59.327Z',
+            password
+        };
+        token = new TokenModel(createConfig);
+    });
+
+    afterEach(() => {
+        datastore = null;
+        mockery.deregisterAll();
+        mockery.resetCache();
+    });
+
+    after(() => {
+        mockery.disable();
+    });
+
+    it('is constructed properly', () => {
+        assert.instanceOf(token, TokenModel);
+        assert.instanceOf(token, BaseModel);
+        schema.models.token.allKeys.forEach((key) => {
+            assert.strictEqual(token[key], createConfig[key]);
+        });
+    });
+
+    describe('update', () => {
+        beforeEach(() => {
+            ironMock.seal.yieldsAsync(null, 'sealed');
+        });
+
+        it('promises to update a token and seal the value before datastore saves it', () => {
+            token.value = 'unsealed';
+
+            return token.update()
+                .then(() => {
+                    assert.calledWith(ironMock.seal, 'unsealed', password, ironMock.defaults);
+
+                    assert.isTrue(datastore.update.calledWith({
+                        table: 'tokens',
+                        params: {
+                            id: 6789,
+                            value: 'sealed',
+                            lastUsed: 'updated timestamp'
+                        }
+                    }));
+                });
+        });
+
+        it('updates the lastUsed time when a token is updated with no fields changed', () => {
+            assert.isFalse(token.isDirty());
+
+            // eslint-disable-next-line no-global-assign
+            Date = function Date() {
+                this.toISOString = () => 'a more different timestamp';
```

Comment:
Was a reference to an internet video, not great for readability though I guess :P